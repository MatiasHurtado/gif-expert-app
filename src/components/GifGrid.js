// import {} from "react"
import React from "react";
import PropTypes from 'prop-types';
import { useFetchGifs } from "../hooks/useFetchGifs";
import GifGridItem from "./GifGridItem";
// import { getGifs } from "../helpers/getGift"

export default function GifGrid({ category }) {
  //Cuando le ponemos : es por que le estamos cambiando el valor al arreglo
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>

      {loading && <p>Loading....</p>}
      <div className ="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
}


GifGrid.propTypes ={
  category: PropTypes.string.isRequired
}