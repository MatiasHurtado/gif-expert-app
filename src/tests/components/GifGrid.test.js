import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from "enzyme"
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas a GifGrid', () => {
   
   const category ='One Punch'



   test('Debe Mostrarse GifGrid Correctamente', () => {

      useFetchGifs.mockReturnValue({
         data: [],
         loading: true
      })

      const wrapper = shallow(<GifGrid category={category}/>)
      expect(wrapper).toMatchSnapshot();
      
   })

   test('Debe de mostrar items cuando se catgan las imagenes en UseFetchGifs', () => {
      
      const gifs =[{
         id:'112',
         url:'https://localHost/aaaaa',
         title:'Cualquier Cosa'
      }]
      
      useFetchGifs.mockReturnValue({
         data: gifs,
         loading: false
      })

      const wrapper = shallow(<GifGrid category={category}/>)
      //Forma Facil de ver si nuestro componente funciona perfectamente

      expect(wrapper.find('p').exists()).toBe(false);
      expect(wrapper.find('GifGridItem').length).toBe(gifs.length)

   })
   
   

})
