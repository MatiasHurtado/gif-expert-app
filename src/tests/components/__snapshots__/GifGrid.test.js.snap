// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a GifGrid Debe Mostrarse GifGrid Correctamente 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <p>
    Loading....
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
